### DISCUSSION TITLE
Question about the off grid beat recording.

### DISCUSSION THREAD

**Discussion by ljubomirpapuga** [bangcorrupt/hacktribe#36] [Ideas]
First of all, thanks for the hard work! 

Second of all, I know I might be asking impossible here, but... 

Currently, when you live record the beat, your beats are recorded strictly to the 16 step grid. 

You can do a lot with grooves, but is there any possibility to turn off grid recording (quantization if you will) and record off the grid?

**Comment by bangcorrupt** [comment]
I hope so.  Gate time is recorded live with 96 ticks per step, so most of the function is definitely already there.  I haven't managed to find the right instructions yet.

**Comment by BKLronin** [comment]
This is my most wanted feature for the E2, I didn´t even asked for it as it seemed so hardcoded.

**Reply by bangcorrupt** [reply]
I don't know if it will ever work.  If it does, we could use groove templates in reverse as a scaled quantize.

**Reply by BKLronin** [reply]
Like writing a temporary groove pattern for the recorded pad based on the ticks offset value?

It´s a bit suspicious that they shorten the bar length when switching to 32th. I tried 32th but it still wasn´t enough for fine grooves.

**Comment by ptowns** [comment]
Omg! Thank you so much for your work!
 So for off grid recording my work around is to 
A) resample and play off grid, ala sp404 style. Will make a loop sample of say you playing high hats off grid. 
B) record a sample with a few milliseconds of silence before them and then use the edit knob to pull them on grid.. confusing and takes some playing with also messes up the amp and filter EG the further off grid you go.

**Comment by BKLronin** [comment]
After some more experimenting, this is still the most important missing feature for me.

**Comment by ecco00** [comment]
woohoo by halving the ticks (therefore doubling the bars available) we could have at least 256 steps.

**Comment by maks** [comment]
I've recently found the notes about the event recorder format and since each event in the recording format just has a millisecond time offset, I'm guessing the event recorder can be used to achieve off grid already with just the standard firmware?

**Reply by bangcorrupt** [reply]
Yes, although at the moment only playback is supported.  I haven't looked at hacking yet, but it could make a really powerful song mode if it had loop points and overdubbing.

**Reply by maks** [reply]
I'm just sticking to using the existing pattern data stuctures because I can read/write them over midi, but if it was possible to read/write the event recorder files over midi they would make a better (though more work to implement)output target for my "tracker app" than the standard patterns.

**Comment by bogdanoctav** [comment]
Hey! I happened to suggest the same things.
Some inspiration on how to approach this might be the substeps approach on the Novation Circuit, and as far as live recording, there is the recording mode on the MPCs or on the Roland spd-30. In that you select quantization values or unquantized recording. They all work well.
Something like that would render the electribe so much more organic.

Agree to the others, the event recorder seems to be able to record in an unquantized way, or at a quantization that is fine enough to not feel the grid too much.

Bangcorrupt, if you're able to pull this off it would be more than awesome.

**Comment by scottishcammy** [comment]
Hi, thanks to all for the work on this. I just bought the Redtribe. I wonder, is removing quantisation still being worked on?

**Reply by bangcorrupt** [reply]
I do plan to work on this, but it may be a while before I can.  I have a way to edit groove templates, which could be used to manually shift steps away from the grid, I'll work on making this more user friendly.

**Comment by e2887341-cloud** [comment]
Disabling quantization is essentially the biggest fly in the ointment for me. If you were to solve this issue in future updates, I believe Korg would be obligated to award you a bonus for your contributions to their product.

**Comment by vanana-old-acc** [comment]
Technical:  I wonder if quantization happens because note positional offset is scarcely represented into quantized values or that it makes use of it's full storage size but there's simply no option to disable quantizing during recording

**Reply by BKLronin** [reply]
Its not quantization it just falls in the nearesr position I think. 

The only way would be to save the offset from when the trigger happens but you can't hack precise time measurement in even if you find a place to dave that information with the note.

So from what I understand, it's impossible.

**Reply by bangcorrupt** [reply]
Part of the sequencing code counts through 96 ticks per step to interpolate the modulation and calculate the groove template timing.  In theory, groove timing offset could be recorded in real time and saved in the pattern data with the note trigger, but we would still only get one note on event per step.  Hacking extra branches into the inner loops of the sequencer code is unlikely to reduce timing bugs.
